### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by engineersmnky** [westonganger/fast_try#1]
This part of the Documentation is incorrect:

>"Apparently safe navigation does not work with class methods so this does not either. Major bummer."

Safe Navigation operator works just fine for Class instance methods 

    Class&.to_s&.class&superclass&.name&.upcase
    #=> "OBJECT"

**Comment by westonganger** [comment]
Sorry I guess I need to change the description of that limitation. It appears to break only when the method does not exist on the class. For example:

```ruby
Array&.bad_method 
#=> NoMethodError: undefined method `bad_method' for Array:Class
```

Let me know if you know how to get around this without killing any performance

**Comment by engineersmnky** [comment]
There is nothing you can do without impacting performance of the safe operator because all additions would require method calls. I think the simplest solution if you really intended to make it more similar to `ActiveSupport#try` would be: 

    self&.public_send(a,*b,&c) if respond_to?(a) 

This nearly identical to how `ActiveSupport` handles it in `try`

**Comment by westonganger** [comment]
Yeah that was the only other solution I had determined. However then it will no longer behave like safe navigation (and I dont think it needs the safe navigation anymore either).

**Comment by engineersmnky** [comment]
the safe navigation only offers assistance to `nil` so it would work in the identical fashion it currently does excepting the fact that it will return `nil` if the object does not respond to the method name. e.g. 

    def simple_test(obj,method_name)
        if obj.respond_to?(method_name)
           puts "#{obj.inspect} responds to #{method_name}"
           obj&.public_send(method_name) 
       end
    end
  

    simple_test(nil,:to_s) 
    # nil responds to to_s
    #=> nil 
    simple_test("a",:to_s)
    # a responds to to_s
    #=> "a"
    simple_test(
         simple_test("a",:non_existent) # this will return nil
    ,:to_s)
    # nil responds to to_s
    #=> nil

**Comment by westonganger** [comment]
Hmm interesting, however `to_s` is a bad use case for this method but its a limitation none-the-less.

Given all the differences it seems I may change the name of this repo to something else.
